corehq/apps/userreports/tests/test_report_filters.py

```diff
@@ -784,6 +807,75 @@ def test_filter_value(self):
         }
         filter_value = LocationDrilldownFilterValue(filter, ['Middlesex'])
         self.assertDictEqual(filter_value.to_sql_values(), {'block_id_drill_0': 'Middlesex'})
+        self.assertEqual(
+            str(filter_value.to_sql_filter().build_expression(mock_table)),
+            'block_id IN (:block_id_drill_0)'
+        )
+        self.assertEqual(
+            filter_value.to_sql_values(),
+            {'block_id_drill_0': 'Middlesex'}
+        )
+
+    def test_ancestor_expression(self):
+        with self.assertRaises(BadSpecError):
+            ReportFilterFactory.from_spec({
+                "type": "location_drilldown",
+                "field": "block_id",
+                "slug": "block_id_drill",
+                "display": "Drilldown by Location",
+                "ancestor_expression": {
+                    'field': 'state_id',
+                    # missing 'location_type': 'state',
+                }
+            })
+
+    def test_prefix_ancestor_location(self):
+        from sqlalchemy import Column, String
+        mock_table = Mock()
+        mock_table.c = [Column(f, type_=String) for f in ['block_id', 'state_id']]
+
+        filter = {
+            "type": "location_drilldown",
+            "field": "block_id",
+            "slug": "block_id_drill",
+            "display": "Drilldown by Location",
+            "ancestor_expression": {
+                'field': 'state_id',
+                'location_type': 'state',
+            }
+        }
+        middlesex_id = self.locations['Middlesex'].location_id
+        mass_id = self.locations['Massachusetts'].location_id
+        # make sure ancestor gets passed if right block is passed
+        filter_value = LocationDrilldownFilterValue(filter, [middlesex_id])
+        self.assertEqual(
+            str(filter_value.to_sql_filter().build_expression(mock_table)),
+            'state_id = :state_id AND block_id IN (:block_id_drill_0)'
+        )
+        self.assertEqual(
+            filter_value.to_sql_values(),
+            {'state_id': mass_id, 'block_id_drill_0': middlesex_id}
+        )
+        # make sure ancestor doesn't get passed if multiple locations are passed
```

Comment:
This actually might be useful. How hard would it be to look up all ancestors of the locations passed and append them to the list like `state_id IN (state1, state2)`?

Reply:
Below are couple of reasons I decided to go this way, 
- There is no way of bulk looking up ancestors of multiple locations. So there will be as many queries as the number of selections
- There is no guarantee if the query will fall into a single partition if there are multiple selections
- In practicality, I thought users won't be selecting multiple locations as often.

Reply:
It is a bit tricky to implement but not impossible. Let me know if there are other compelling reasons to do it.

Reply:
I think given these reasons it makes sense not to implement this. Thanks for listing those. It may be worthwhile to investigate later, but not a high ROI now. Putting in some more context here:

> There is no guarantee if the query will fall into a single partition if there are multiple selections

FWIW this could still be helpful for instance if there a user selects two districts in two different states as it would only need to look at 2 tables instead of 36.

> In practicality, I thought users won't be selecting multiple locations as often.

For ICDS I think there are still some instances of reports in the old UI where this can happen, but as we move more reports to the dashboard this won't matter.